app/uk/gov/hmrc/apipublisher/controllers/PublisherController.scala

```diff
@@ -55,17 +56,45 @@ class PublisherController @Inject()(publisherService: PublisherService,
 
   private def publishService(serviceLocation: ServiceLocation)(implicit hc: HeaderCarrier): Future[Result] = {
     Logger.info(s"Publishing service $serviceLocation")
-    publisherService.publishAPIDefinitionAndScopes(serviceLocation).map {
-      case Some(true) =>
-        Logger.info(s"Successfully published API Definition and Scopes for ${serviceLocation.serviceName}")
-        NoContent
-      case Some(false) =>
-        Logger.info(s"Publication awaiting approval for ${serviceLocation.serviceName}")
-        Accepted
-      case None =>
-        Logger.info(s"API publishing disabled for ${serviceLocation.serviceName}")
-        NoContent
-    } recover recovery(s"$FAILED_TO_PUBLISH ${serviceLocation.serviceName}")
+
+    import cats.data.{OptionT, EitherT}
+    import cats.implicits._
+
+    type Over[A] = EitherT[Future,Result, A]
+
+    def getDefinition: Over[ApiAndScopes] = {
+      EitherT(definitionService.getDefinition(serviceLocation).map(_.toRight(BadRequest)))
+    }
+
+    def validate(apiAndScopes: ApiAndScopes): Over[Unit] = {
+      EitherT.fromOptionF(
+        OptionT(publisherService.validateAPIDefinitionAndScopes(apiAndScopes)).map { e =>
+          Console.println("***** "+e)
```

Comment:
Should we use the logger?

Reply:
I've removed it altogether - was only in use for a debug.